routers/api/v1/api.go

```diff
@@ -576,7 +576,7 @@ func Routes(sessioner func(http.Handler) http.Handler) *web.Route {
 		// Miscellaneous
 		if setting.API.EnableSwagger {
 			m.Get("/swagger", func(ctx *context.APIContext) {
-				ctx.Redirect("/api/swagger")
+				ctx.Redirect(setting.AppURL + "api/swagger")
```

Comment:
Must `setting.AppURL` have the trailing slash or not?
I am thoroughly confused by now.

Reply:
Take a look at #12999

https://github.com/go-gitea/gitea/blob/58cd55d3535e5afe551610cb74588b33d97786a2/modules/setting/setting.go#L78-L80

Reply:
https://github.com/go-gitea/gitea/blob/58cd55d3535e5afe551610cb74588b33d97786a2/modules/setting/setting.go#L81-L83

Reply:
I always confuse these two when reviewing :angry: 
Ignore both my comments then.

Reply:
Although I think we've been changing most of our urls to:

```suggestion
				ctx.Redirect(setting.AppSubURL + "/api/swagger")
```

Reply:
Thanks zeripath. Yes, the code is correct. But I decide to use AppSubURL instead of AppURL in api/v1 now, because all HTTP client can handle relative path correctly and we do not need to use AppURL here.

PR updated.

Reply:
Oh, just saw your comment, yes, I have changed all AppURL in api/v1 to AppSubURL. 😊